src/messaging/ExchangeMgr.h

```diff
@@ -48,6 +48,8 @@ static constexpr int16_t kAnyMessageType = -1;
  */
 class DLL_EXPORT ExchangeManager : public SecureSessionMgrDelegate
 {
+    friend class MessageCounterSyncMgr;
```

Comment:
why is a friend class necessary? Can it be avoided?

Friend classes often indicate tight coupling.

Reply:
Like ReliableMessageMgr, MessageCounterSyncMgr is separated from ExchangeContext/ExchangeMgr logically, MessageCounterSyncMgr needs to call into the private functions of  ExchangeContext/ExchangeMgr, they are tightly coupled. I removed the friend class declaration in ExchangeMgr by using an intermediate function, but keep the friend declaration in ExchangeContext since MessageCounterSyncMgr need to access the private SendMessage of ExchangeContext.